db/migrate/20191016154136_rsvp_migration.rb

```diff
@@ -0,0 +1,25 @@
+class RsvpMigration < ActiveRecord::Migration[6.0]
+  def change
+    custom_fields = TopicCustomField.where(name: 'event_going')
+    custom_fields.each do |custom_field|
+      going = custom_field.value.split ','
+      next if going.empty?
+      going_ids = []
+
+      going.each do |v|
+        next if v.empty? || v.to_i != 0 #its an integer in quotes so its either an event created using latest code or already migrated
+        begin
+          user_id = User.find_by(username: v).id
+        rescue ActiveRecord::RecordNotFound => e
+          print "invalid username #{v}" #safely ignored invalid user id
+        else
+          going_ids.push user_id
+        end
+      end
+
+      custom_field.value = going_ids.to_json
```

Comment:
@fzngagan Can you remind me why we're casting this to JSON when the data is changing from this

```
angus,fzngagan
```
To this
```
1,4
```

i.e. from a comma seperated list of usernames to a comma seperated list of ids. Why are we changing this to JSON?

Also, the way we're saving the data in ``PostRevisor.track_topic_field(:event)`` hasn't changed. It's still just a string of ids.

```
going = event['going'] ? event['going'].join(',') : ''
```
I'm not confident we've got the data validations / types right yet.

Can you test all the different ways of adding RSVPs (event admin panel, user rsvp), and also double check this migration (i.e. adding data before the changes, then migrating the old data. 

We need to be 100% sure the data is handled correctly in the migration.

Reply:
We did decide to convert this to a JSON array. Hence the conversion. The reason being, parsing it gives us the correct data type and not strings.

I missed changing the logic in PostRevisor though. I'll update it and ping...

Reply:
@angusmcleod I'm thinking whether updating the rsvp stuff should be in the post revisor. I think the rsvp data is being handled through the rsvp controller so do we need to update rsvp data in the post revisor. Isn't it redundant?